publish/girder.py

```diff
@@ -62,3 +73,16 @@ def iter_file_content(self, file_id: str) -> Iterator[bytes]:
         with self.stream('GET', f'file/{file_id}/download') as resp:
             resp.raise_for_status()
             yield resp.iter_bytes()
+
+
+@contextlib.contextmanager
+def dandiset_lock(dandiset_id, client: GirderClient):
+    resp = client.post(f'dandi/{dandiset_id}/lock')
+    if (resp.status_code != 200):
```

Comment:
Normally, I think `raise_for_status` would be more idiomatic than `resp.status_code`, but since we expect exactly a 200 (other 2xx success codes would still be unacceptable), this is better.